analyzers/tests/SonarAnalyzer.UnitTest/TestCases/OperatorsShouldBeOverloadedConsistently.CSharp11.cs

```diff
@@ -0,0 +1,13 @@
+﻿public class Foo : IEquality<Foo> // Noncompliant FP Repro for: https://github.com/SonarSource/sonar-dotnet/issues/6442
+{
+    public static Foo operator +(Foo a, Foo b) => new Foo();
+
+    public override bool Equals(object obj) => false;
+    public override int GetHashCode() => 0;
+}
+
+public interface IEquality<TSelf> where TSelf : IEquality<TSelf>
+{
+    static virtual bool operator ==(TSelf a, TSelf b) => true;
+    static virtual bool operator !=(TSelf a, TSelf b) => false;
+}
```

Comment:
Reminder to myself: Probably not an FP. The operators are only accessible via constrained generic parameters. Need to check later.

Reply:
I thought about this a bit and I think we should make sure, that we don't raise on operators on interfaces. The reasoning is that you want to use the interfaces to describe the operators supported by a type in a fine grained way. e.g. [IAdditionOperators<TSelf,TOther,TResult> Interface (System.Numerics)](https://learn.microsoft.com/en-us/dotnet/api/system.numerics.iadditionoperators-3?view=net-7.0) does not derive from [IEqualityOperators<TSelf,TOther,TResult> Interface (System.Numerics)](https://learn.microsoft.com/en-us/dotnet/api/system.numerics.iequalityoperators-3?view=net-7.0) and vice versa. Please add tests for operators in interfaces and make sure we don't raise for it:
```cs
public interface IPlus where TSelf: IPlus<TSelf>
{
    static virtual TSelf operator +(TSelf a, TSelf b) => a;
}
public interface IMinus where TSelf: IMinus<TSelf>
{
    static abstract TSelf operator -(TSelf a, TSelf b);
}
public interface IMulti
{
    static IMulti operator *(IMulti a, IMulti b) => a;
}
```

Reply:
I added the examples but [the rule registers only for](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.CSharp/Rules/OperatorsShouldBeOverloadedConsistently.cs#L67) ` SyntaxKind.ClassDeclaration`, so we should be safe.